### ISSUE TITLE
Permanently disabled holderName after focus on SF input field on Mobile Safari

### ISSUE THREAD

**Issue by nterziyski** [Adyen/adyen-web#1722]
**Describe the bug**
On Mobile Safari browsers on touch start of secure-field input, non-secure-field inputs get "disabled" (as described [here](https://github.com/Adyen/adyen-web/blob/951dc7cabdca1e8b2f3cc55d25be6cbf181f7361/packages/lib/src/components/Card/components/CardInput/CardInput.tsx#L143-L145)). Unfortunately currently there is no "reset" happening on focus of non-SF inputs, leading to non-SF inputs such as the holderName being disabled forever and preventing users from submitting their payment details in case holderNameRequired is set to true.

Currently used adyen-web package version: 5.18.0

Here is also the used paymentMethodsConfiguration:

```
paymentMethodsConfiguration: {
          card: {
            hasHolderName: true,
            holderNameRequired: true,
            enableStoreDetails: true,
            name: 'Credit or debit card'
          },
          paypal: {
            blockPayPalCreditButton: true,
            blockPayPalPayLaterButton: true,
          },
},
```

**To Reproduce**
1. Go to an instantiated drop-in context with version 5.18.0 or later
2. Make sure the drop-in is setup to have hasHolderName and holderNameRequired set as true
3. Click on Credit Card
4. Input test CC number (e.g. 4000 0200 0000 0000)
5. Input expiry date (e.g. 03/30)
6. The holderName input field node node will be disabled (as shown on the included screenshot)
7. You won't be able to focus/input any data in the holderName and hence also be unable to submit the form

**Expected behavior**
I expect that I would be able to provide credit card details, including card holder name, and be able to submit the form. 

A potential code change idea would be to either make use of an onBlur event to reset the focused iOSFocusedField prop and/or give non-SF inputs the same onTouchStart behaviour

**Screenshots**
<img width="1442" alt="Screenshot 2022-09-06 at 15 23 20" src="https://user-images.githubusercontent.com/24291601/188646785-5178cdc0-f244-468d-a238-df5677831fa2.png">

**Desktop (please complete the following information):**
 - OS: MacOS Catalina
 - Browser: Browser Safari with Mobile UA
 - Version: 13.1.3 / 14.0

**Smartphone (please complete the following information):**
 - Device: iPhone 13 mini
 - OS: 15.6.1
 - Browser: Mobile Safari
 - Version: 15.6

**Comment by sponglord** [comment]
Hi @nterziyski - thanks for raising this issue. 

So far, however, I have not been able to recreate it _except_ in Safari with Responsive Design Mode.

I have tried on the following _actual_ iPhones:
- iPhone 6 v12.4.7
- **iPhone 13 mini v15.6.1**
- iPhone 12 pro v15.6.1

Also in the XCode Simulator for:
- iPhone SE v15.5

And in Browserstack on the following:
- iPhone 8 v11.4
- iPhone 8 v12.1
- iPhone 8 v13.2
- iPhone 8 v15.4
- iPhone 13 mini v15.3
- iPhone 13 pro v15.6
- iPhone SE v11.2
- iPhone 6 v8.0

I _can_ recreate the issue in Safari with Responsive Design Mode for:
- iPhone 8
- iPhone 8+
- iPhone SE
- iPad mini
but _not_ for:
- iPad
- iPad Pro (10.5")
- iPad Pro (12.9")

This was all tested against a url running Dropin v5.23.1. 
However the change that causes this behaviour was made in v5.15.0 and no changes related to this functionality have been made since.

Do you have url I can test against?

**Comment by nterziyski** [comment]
Thank you for looking into this @sponglord!

Unfortunately we have downgraded our live implementation to v5.14.0 yesterday (after looking through the code, we figured that the behaviour was introduced in v5.15.0 and as you pointed out, nothing has changed in the behaviour since then) and the people that can whitelist a sandbox domain for our test key are out until Friday.

I can confirm that we discovered the issue on Desktop, in Safari, with Responsive Design Mode. But we have proceeded to also reproduce it with v5.18.0 running live on physical iOS devices.

**Something to note** is that the card holder input field gets disabled after entering the expiry date and the system switches/steals the focus to the CVC/CVV input. 

Additionally, I can confirm your findings that the issue is not present in virtualised iPhones in Browserstack, as well as that it is not present on Safari with Responsive Design Mode for iPads (except for iPad mini). I will try to test a local deployment on iPad Pro next and will come back with a status update.

**Comment by nterziyski** [comment]
OK, we managed to create an isolated sandbox and to cover more tests with real device. I can hereby confirm your findings that the issue is **not** reproducible on the following real devices:

- iPad Pro (10.5") v14.4 (Safari v14.1.2)
- iPhone 12 mini v14 (Safari v14.0)
- iPhone 12 mini v15.6.1 (Safari v15.6)
- iPhone 13 mini v15.6.1 (Safari v15.6)

and thus seems that the issue is isolated to Safari on Desktop in Responsive Design Mode.

Closing the issue and changing the title to clarify the impact scope for future adventurers that stumble upon it.
